pyro/contrib/autoguide/__init__.py

```diff
@@ -306,7 +306,11 @@ def __call__(self, *args, **kwargs):
                 value = pyro.param("{}_{}".format(self.prefix, name),
                                    site["value"].detach(),
                                    constraint=site["fn"].support)
-                result[name] = pyro.sample(name, dist.Delta(value, event_dim=site["fn"].event_dim))
+                transform = biject_to(site["fn"].support)
```

Comment:
@fritzo had to add this stuff. besides letting this log det jacobian logic be controlled by a flag (`include=True/False`) i could instead introduce a subclass:
```class SVGDGuide(AutoDelta):....```